concatenate_ice/concat_ice_daily.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/bash
+# Copyright 2024 ACCESS-NRI and contributors. See the top-level COPYRIGHT file for details.
+# SPDX-License-Identifier: Apache-2.0
+# 
+# Concatenate sea-ice daily output from access-om2
+# this was written assuming it would be used as a payu "userscript" at the "archive" stage, but alternatively a path to an "archive" directory can be provided
+# Script inspired from https://github.com/COSIMA/1deg_jra55_ryf/blob/master/sync_data.sh#L87-L108
+#
+# This script uses "ncrcat". Load this through either 'module use /g/data/vk83/modules; module load payu' or 'module load nco'.
+
+Help()
+{
+   # Display Help
+   echo "Concatenante daily history output from the (sea) ice model to a single file"
+   echo
+   echo "Syntax: scriptTemplate [-h|d DIRECTORY]"
+   echo "options:"
+   echo "h     Print this Help."
+   echo "d     Process "name" directory rather than latest output in archive folder."
+   echo
+}
+
+# Get the options
+while getopts ":hd:" option; do
+   case $option in
+      h) # display Help
+         Help
+         exit;;
+      d) # Enter a directory
+         out_dir=$OPTARG/ice/OUTPUT
+         if [ ! -d $out_dir ]; then 
+            echo $out_dir Does not exist
+            exit
+         fi;;  
+     \?) # Invalid option
+         echo "Error: Invalid option"
+         exit;;
+   esac
+done
+
+#If no directory option provided , then use latest
+if [ -z $out_dir ]; then 
+    #latest output dir only
+    out_dir=$(ls -drv archive/output??? | head -1)/ice/OUTPUT 
```

Comment:
```suggestion
    out_dir=$(ls -drv archive/output*[0-9] | head -1)/ice/OUTPUT 
```
the `archive/output???` glob only matches 3 digits. As we need to match 3 or 4, without using a regex extension for globbing I think the above suggestion is the best we can do.

Reply:
Heh. I got the above from this [negged SO comment](https://stackoverflow.com/a/63222611/4727812). Now I'm not sure how standard it is, specifically the use of parentheses.

Reply:
Yeah that needs extended globbing. So either `shopt -x extglob` or `archive/output[0-9][0-9][0-9]*`.

If you do go with `extglob` then `output[0-9][0-9][0-9]*([0-9])` is even more specific.

Reply:
oops - sorry. I tested just ` archive/output*` but forgot to make the change! 

Anyway, I went we extglob for completeness/robustness.